lib/src/presentation/bloc/fetch_movies_bloc_impl.dart

```diff
@@ -0,0 +1,55 @@
+import 'dart:async';
+
+import '../../core/bloc/i_fetch_movies_bloc.dart';
+import '../../core/util/data_state.dart';
+import '../../core/util/enum_classes.dart';
+import '../../data/model/movie_model.dart';
+import '../../domain/entity/movie_state.dart';
+import '../../domain/usecase/usecase_impl/get_movies_usecase_impl.dart';
+
+class FetchMoviesBloc extends IFetchMoviesBloc {
+  final StreamController<MovieState> _moviesFetcher =
+      StreamController<MovieState>.broadcast();
+  final GetMoviesUseCase getMoviesUseCase;
+
+  FetchMoviesBloc({required this.getMoviesUseCase});
+
+  @override
+  void dispose() {
+    _moviesFetcher.close();
+  }
+
+  @override
+  void fetchMovieByCategory(String category) async {
+    _moviesFetcher.sink.add(
+      MovieState.loading(),
+    );
+    DataState<MovieModel> movieState =
+        await getMoviesUseCase.getMovieModel(category);
+    if (movieState.state == BaseState.success) {
+      if (movieState.data != null) {
+        _moviesFetcher.sink.add(
+          MovieState.success(
+            movieList: movieState.data!.result,
+          ),
+        );
+      } else {
+        _moviesFetcher.sink.add(
+          MovieState.empty(),
+        );
+      }
```

Comment:
You can replace this kind of validation with this
`_moviesFetcher.sink.add(movieState.data != null ? MovieState.success(...) : MovieState.empty())`

Reply:
Review the name of the controller, instead to _moviesFetcher is a _movieController, also review the name of the class Fetch is an action, and the class name shouldn't express that